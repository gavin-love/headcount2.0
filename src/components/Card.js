import React from "react";
import "../styles/Card.css";
import PropTypes from "prop-types";

const Card = props => {
  if (props.stats) {
    const districtValues = Object.values(props.stats);
    const districtStat = districtValues.map((stat, index) => {
      if (stat >= 0.5) {
        return (
          <li className="greater" key={index}>
            {stat}
          </li>
        );
      } else {
        return (
          <li className="lessThan" key={index}>
            {stat}
          </li>
        );
      }
    });

    return (
      <div onClick={() => props.findAverages(props)} className="card">
        <h1>{props.location}</h1>
        <ul>{districtStat}</ul>
      </div>
    );
  } else if (!props.stats) {
    console.log(props);
    return (
      <div
        onClick={() => props.findAverages(props)}
        className="compare-card card"
      >
        <h1>{props.locationOne}</h1>
        <h1>{props.locationOneAverage}</h1>
        <p>{props.comparedAverage}</p>
        <h1>{props.locationTwo}</h1>
        <h1>{props.locationTwoAverage}</h1>
      </div>
    );
  }
};

Card.propTypes = {
  stats: PropTypes.object,
  location: PropTypes.string,
  locationOne: PropTypes.string,
  locationOneAverage: PropTypes.number,
  locationTwoAverage: PropTypes.number,
  comparedAverage: PropTypes.number,
  locationTwo: PropTypes.string
};

export default Card;
