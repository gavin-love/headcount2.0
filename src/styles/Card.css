.greater {
  color: red
}

.lessThan {
  color: blue;
}