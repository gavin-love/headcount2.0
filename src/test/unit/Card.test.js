import React from "react";
import Card from "../../components/Card.js";
import { shallow } from "enzyme";

describe("Card", () => {
  let wrapper;
  let mockStats;
  let mockFunction;

  beforeEach(() => {
    mockStats = { 2004: 0.24, 2005: 0.63, 2006: 0.11 };
    mockFunction = jest.fn();
  });

  it("matches the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should return a stats card when passed an object with property of stats", () => {
    wrapper = shallow(
      <Card findAverages={mockFunction} location="colorado" stats={mockStats} />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("should return a stats card when passed an object with averages", () => {
    wrapper = shallow(
      <Card
        locationOne="COLORADO"
        locationOneAverage={0.53}
        comparedAverage={0.46}
        locationTwo="ACADEMY 20"
        locationTwoAverage={0.4}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
