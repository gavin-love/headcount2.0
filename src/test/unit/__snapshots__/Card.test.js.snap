// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card matches the snapshot 1`] = `undefined`;

exports[`Card should return a stats card when passed an object with averages 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    comparedAverage={0.46}
    locationOne="COLORADO"
    locationOneAverage={0.53}
    locationTwo="ACADEMY 20"
    locationTwoAverage={0.4}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          COLORADO
</h1>,
        <h1>
          0.53
</h1>,
        <p>
          0.46
</p>,
        <h1>
          ACADEMY 20
</h1>,
        <h1>
          0.4
</h1>,
      ],
      "className": "compare-card card",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "COLORADO",
        },
        "ref": null,
        "rendered": "COLORADO",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 0.53,
        },
        "ref": null,
        "rendered": 0.53,
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 0.46,
        },
        "ref": null,
        "rendered": 0.46,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "ACADEMY 20",
        },
        "ref": null,
        "rendered": "ACADEMY 20",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 0.4,
        },
        "ref": null,
        "rendered": 0.4,
        "type": "h1",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            COLORADO
</h1>,
          <h1>
            0.53
</h1>,
          <p>
            0.46
</p>,
          <h1>
            ACADEMY 20
</h1>,
          <h1>
            0.4
</h1>,
        ],
        "className": "compare-card card",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "COLORADO",
          },
          "ref": null,
          "rendered": "COLORADO",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 0.53,
          },
          "ref": null,
          "rendered": 0.53,
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 0.46,
          },
          "ref": null,
          "rendered": 0.46,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "ACADEMY 20",
          },
          "ref": null,
          "rendered": "ACADEMY 20",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 0.4,
          },
          "ref": null,
          "rendered": 0.4,
          "type": "h1",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Card should return a stats card when passed an object with property of stats 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    findAverages={[Function]}
    location="colorado"
    stats={
        Object {
            "2004": 0.24,
            "2005": 0.63,
            "2006": 0.11,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          colorado
</h1>,
        <ul>
          <li
                    className="lessThan"
          >
                    0.24
          </li>
          <li
                    className="greater"
          >
                    0.63
          </li>
          <li
                    className="lessThan"
          >
                    0.11
          </li>
</ul>,
      ],
      "className": "card",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "colorado",
        },
        "ref": null,
        "rendered": "colorado",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="lessThan"
>
              0.24
</li>,
            <li
              className="greater"
>
              0.63
</li>,
            <li
              className="lessThan"
>
              0.11
</li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": 0.24,
              "className": "lessThan",
            },
            "ref": null,
            "rendered": 0.24,
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": 0.63,
              "className": "greater",
            },
            "ref": null,
            "rendered": 0.63,
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "host",
            "props": Object {
              "children": 0.11,
              "className": "lessThan",
            },
            "ref": null,
            "rendered": 0.11,
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            colorado
</h1>,
          <ul>
            <li
                        className="lessThan"
            >
                        0.24
            </li>
            <li
                        className="greater"
            >
                        0.63
            </li>
            <li
                        className="lessThan"
            >
                        0.11
            </li>
</ul>,
        ],
        "className": "card",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "colorado",
          },
          "ref": null,
          "rendered": "colorado",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="lessThan"
>
                0.24
</li>,
              <li
                className="greater"
>
                0.63
</li>,
              <li
                className="lessThan"
>
                0.11
</li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": 0.24,
                "className": "lessThan",
              },
              "ref": null,
              "rendered": 0.24,
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": 0.63,
                "className": "greater",
              },
              "ref": null,
              "rendered": 0.63,
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "host",
              "props": Object {
                "children": 0.11,
                "className": "lessThan",
              },
              "ref": null,
              "rendered": 0.11,
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
